src/utils.js

```diff
@@ -99,15 +99,30 @@ export function addUserToHTPasswd(
  * @param {number} maxUsers
  * @returns {object}
  */
-export function sanityCheck(user: string, users: {}, maxUsers: number) {
+export function sanityCheck(
+  user: string,
+  password: string,
+  verifyFn: Function,
+  users: {},
+  maxUsers: number
+) {
   let err = null;
+  // auth returns boolean
   if (users[user]) {
-    err = Error('this user already exists');
+    let auth = verifyFn(password, users[user]);
+    if (auth) {
+      return err;
```

Comment:
Just noticed a bug here. It will make an entry in htpasswd file on successful auth.

Reply:
Yeah,  the logic is a bit odd. I was trying to match the code with the test

Reply:
It's good we have plenty of unit test, that will help to avoid issues in `verdaccio` since this is the default auth plugin.